src/environmentVariables.ts

```diff
@@ -0,0 +1,147 @@
+import * as os from 'os';
+import * as util from 'util';
+
+const inspect = util.inspect.custom;
+const envProperty = Symbol('envProperty');
+
+export type EnvironmentVariablesBasic = Record<string, string>;
+export type EnvironmentVariablesNull = Record<string, string | null>;
+export type EnvironmentVariablesUndefined = Record<string, string | undefined>; // alias of NodeJS.ProcessEnv
+export type EnvironmentVariables = Record<string, string | undefined | null>;
+export type EnvironmentVariablesItem = EnvironmentVariables | null | undefined;
+
+class EnvironmentVariablesPrivate {
+    private keyMapping: Map<string, string>;
+    public [envProperty]: EnvironmentVariables;
+    protected isWin32: boolean;
+    protected preserveNull: boolean;
+    constructor(
+        preserveNull: boolean | undefined,
+        isWin32: boolean | undefined) {
```

Comment:
Hi, this is placed here intensionlly, so that we can testing win32/non-win32 on windows and linux without
setup two operating system or waiting the github actions result.
That's simplify the test case a lot, we have no need remove it in the future.